.github/workflows/test.yml

```
@@ -735,3 +735,13 @@ jobs:
           files: ./packages/generator-helper/src/__tests__/coverage/clover.xml
           flags: generator-helper,${{ matrix.os }},library,binary
           name: generator-helper-${{ matrix.os }}
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: ${{ runner.os == 'macos-11' && !cancelled() }} # Run this step even when the tests fail. Skip if the workflow is cancelled.
+        uses: Workshop64/buildpulse-action@master
+        with:
+          account: 17219288
+          repository: 192925833
+          path: .
+          key: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+          secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
```

Following up on the discussion in https://github.com/Workshop64/buildpulse-action/issues/2#issuecomment-1047229576, this change will use the test-reporter binary for macOS (amd64):

```suggestion
        env:
          BUILDPULSE_ACCESS_KEY_ID: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
          BUILDPULSE_SECRET_ACCESS_KEY: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
        run: |
          curl -fsSL https://get.buildpulse.io/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
          chmod +x ./buildpulse-test-reporter
          ./buildpulse-test-reporter submit junit.xml --account-id 17219288 --repository-id 192925833
```

Note that this also updates the path from `.` to `junit.xml`. By default, jest-junit outputs the report to `junit.xml` in the current working directory [[docs](https://github.com/jest-community/jest-junit/blob/v13.0.0/README.md?plain=1#L60-L61)]. By specifying the exact path of the file (as opposed to specifying a directory), we can avoid spending any time looking through the directory structure for XML files. :-)

---

That looks good.

I think we are creating multiple `junit.xml` in different `packages/*` locations though, as the workflow runs tests in different working-directories. Can we make it look in multiple locations at the same time?

---

Oh, thanks for pointing that out. :bow:

In that case, the easiest option is to point to the `packages` directory and have the test-reporter locate the XML files. It'll still be super fast. :-)

```suggestion
        env:
          BUILDPULSE_ACCESS_KEY_ID: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
          BUILDPULSE_SECRET_ACCESS_KEY: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
        run: |
          curl -fsSL https://get.buildpulse.io/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
          chmod +x ./buildpulse-test-reporter
          ./buildpulse-test-reporter submit packages --account-id 17219288 --repository-id 192925833
```